flink-python/src/main/java/org/apache/flink/python/PythonConfig.java

```diff
@@ -19,152 +19,112 @@
-        return maxArrowBatchSize;
-    }
-
-    public Map<String, String> getPythonFilesInfo() {
-        return pythonFilesInfo;
-    }
-
-    public Optional<String> getPythonRequirementsFileInfo() {
-        return Optional.ofNullable(pythonRequirementsFileInfo);
-    }
-
-    public Optional<String> getPythonRequirementsCacheDirInfo() {
-        return Optional.ofNullable(pythonRequirementsCacheDirInfo);
-    }
+    private final ReadableConfig pythonDependencyConfiguration;
 
-    public Map<String, String> getPythonArchivesInfo() {
-        return pythonArchivesInfo;
+    public PythonConfig(ReadableConfig configuration) {
+        this.configuration = Preconditions.checkNotNull(configuration);
+        this.pythonDependencyConfiguration = new Configuration();
     }
 
-    public String getPythonExec() {
-        return pythonExec;
+    public PythonConfig(
+            ReadableConfig configuration, ReadableConfig pythonDependencyConfiguration) {
+        this.configuration = Preconditions.checkNotNull(configuration);
+        this.pythonDependencyConfiguration =
+                Preconditions.checkNotNull(pythonDependencyConfiguration);
     }
 
-    public String getExecutionMode() {
-        return executionMode;
+    @Override
+    public <T> T get(ConfigOption<T> option) {
+        return pythonDependencyConfiguration
+                .getOptional(option)
+                .orElseGet(() -> configuration.get(option));
     }
 
-    public boolean isMetricEnabled() {
-        return metricEnabled;
+    @Override
+    public <T> Optional<T> getOptional(ConfigOption<T> option) {
+        final Optional<T> value = pythonDependencyConfiguration.getOptional(option);
+        if (value.isPresent()) {
+            return value;
+        }
+        return configuration.getOptional(option);
     }
 
-    public boolean isProfileEnabled() {
-        return profileEnabled;
-    }
+    public Configuration toConfiguration() {
+        final Configuration config = new Configuration();
+        PYTHON_CONFIG_OPTIONS.forEach(
+                option ->
+                        getOptional((ConfigOption) option)
+                                .ifPresent(v -> config.set((ConfigOption) option, v)));
+
+        // prepare the job options
+        Map<String, String> jobOptions = config.get(PythonOptions.PYTHON_JOB_OPTIONS);
+        if (jobOptions == null) {
+            jobOptions = new HashMap<>();
+            config.set(PythonOptions.PYTHON_JOB_OPTIONS, jobOptions);
+        }
+        jobOptions.put(
+                PythonOptions.STATE_CACHE_SIZE.key(),
+                String.valueOf(config.get(PythonOptions.STATE_CACHE_SIZE)));
+        jobOptions.put(
+                PythonOptions.MAP_STATE_ITERATE_RESPONSE_BATCH_SIZE.key(),
+                String.valueOf(config.get(PythonOptions.MAP_STATE_ITERATE_RESPONSE_BATCH_SIZE)));
+        jobOptions.put("table.exec.timezone", getLocalTimeZone(configuration).getId());
```

Comment:
Would it be better to use the key of TableConfigOptions.LOCAL_TIME_ZONE directly instead of adding a new table.exec.timezone？